test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingBindingBehaviorTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.Framework.DependencyInjection;
+using ModelBindingWebSite;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    public class ModelBindingBindingBehaviorTest
+    {
+        private const string SiteName = nameof(ModelBindingWebSite);
+        private readonly Action<IApplicationBuilder> _app = new Startup().Configure;
+        private readonly Action<IServiceCollection> _configureServices = new Startup().ConfigureServices;
+
+        [Fact]
+        public async Task BindingBehavior_MissingRequiredProperties_ValidationErrors()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+
+            var url = "http://localhost/BindingBehavior/EchoModelValues";
+            var request = new HttpRequestMessage(HttpMethod.Post, url);
+            var formData = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("model.Property2", "Hi"),
+            };
+
+            request.Content = new FormUrlEncodedContent(formData);
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+            var body = await response.Content.ReadAsStringAsync();
+            var errors = JsonConvert.DeserializeObject<SerializableError>(body);
+
+            Assert.Equal(2, errors.Count);
+
+            var error = Assert.Single(errors, kvp => kvp.Key == "model.Property3" );
+            Assert.Equal("The 'Property3' property is required.", ((JArray)error.Value)[0].Value<string>());
```

Comment:
We may have an issue with `SerializableError` - unfortunately there were no functional tests for using it with Json.Net. Opened a bug to track this.